idea/src/org/jetbrains/kotlin/idea/inspections/RedundantElseInIfInspection.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.codeStyle.CodeStyleManager
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.intentions.branchedTransformations.isElseIf
+import org.jetbrains.kotlin.idea.refactoring.getLineNumber
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.getNextSiblingIgnoringWhitespace
+import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.types.typeUtil.isNothing
+
+class RedundantElseInIfInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        ifExpressionVisitor(fun(ifExpression) {
+            if (ifExpression.isElseIf()) return
+            val elseKeyword = ifExpression.singleElseKeyword(true) ?: return
+            val rangeInElement = elseKeyword.textRange?.shiftRight(-ifExpression.startOffset) ?: return
+            holder.registerProblem(
+                holder.manager.createProblemDescriptor(
+                    ifExpression,
+                    rangeInElement,
+                    "Redundant 'else'",
+                    ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                    isOnTheFly,
+                    RemoveRedundantElseFix()
+                )
+            )
+        })
+}
+
+private class RemoveRedundantElseFix : LocalQuickFix {
+    override fun getName() = "Remove redundant 'else'"
+
+    override fun getFamilyName() = name
+
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
```

Comment:
This fix has a problem: it works incorrectly with long block of statements in else, shifting one tab back only first of them and keeping indent of all other statements. If you like, you can download custom plugin from here: https://teamcity.jetbrains.com/viewLog.html?buildId=1749508&buildTypeId=Kotlin_dev_CompilerAllPlugins&tab=artifacts (beware, after some days artifacts will perish) and test the fix just on Kotlin project (I ran the inspection on just idea module and applied all fixes automatically).